pyproject.toml

```diff
@@ -1,50 +1,57 @@
-[tool.poetry]
+[project]
 name = "linkml-map"
 version = "0.0.0"
 description = "a framework for specifying and executing mappings between data models"
-authors = ["cmungall <cjm@berkeleybop.org>"]
 readme = "README.md"
+authors = [
+    {name = "cmungall", email = "cjm@berkeleybop.org"},
+]
+requires-python = "<=3.13,>=3.9"
```

Comment:
From my PR on updating [linkml-runtime to poetry2](https://github.com/linkml/linkml-runtime/pull/364/). Add:
```
dynamic = ["version"]
```